io/disk/disktest.py

```diff
@@ -133,22 +215,135 @@ def one_disk_chunk(self, disk, chunk):
+        if wait.wait_for(is_raid_deleted, timeout=10):
+            self.log.info("software raid  %s deleted" % self.raid_name)
+        else:
+            self.err_mesg.append("failed to delete sraid %s" % self.raid_name)
+
+    def delete_lv(self):
+        """
+        checks if lv/vg exists and delete them along with its metadata
+        if exists.
+        """
+        def is_lv_deleted():
+            lv_utils.lv_remove(self.vgname, self.lvname)
+            time.sleep(5)
+            lv_utils.vg_remove(self.vgname)
+            if lv_utils.lv_check(self.vgname, self.lvname):
+                return False
+            return True
+        if wait.wait_for(is_lv_deleted, timeout=10):
+            self.log.info("lv %s deleted" % self.lvname)
+        else:
+            self.err_mesg.append("failed to delete lv %s" % self.lvname)
+        # checking and deleteing if lvm_meta_data exists after lv removed
+        cmd = 'blkid -o value -s TYPE %s' % self.disk
+        out = process.system_output(cmd, shell=True,
+                                    ignore_status=True).decode("utf-8")
+        if out == 'LVM2_member':
+            cmd = "wipefs -af %s" % self.disk
+            process.system(cmd, shell=True, ignore_status=True)
+
+    def delete_fs(self, l_disk):
+        """
+        checks for disk/dir mount, unmount if mounted and checks for
+        filesystem exitance and wipe it off after dir/disk unmount.
+
+        :param l_disk: disk name for which you want to check the mount status
+
+        :return: None
+        """
+        def is_fs_deleted():
+            cmd = "wipefs -af %s" % l_disk
+            process.system(cmd, shell=True, ignore_status=True)
+            if disk.is_fs_exists(l_disk):
+                return False
+            return True
+
+        def is_disk_unmounted():
+            cmd = "umount %s" % l_disk
+            cmd1 = 'umount /dev/mapper/avocado_vg-avocado_lv'
+            process.system(cmd, shell=True, ignore_status=True)
+            process.system(cmd1, shell=True, ignore_status=True)
+            if disk.is_disk_mounted(l_disk):
+                return False
+            return True
+
+        def is_dir_unmounted():
+            cmd = 'umount %s' % self.dir
+            process.system(cmd, shell=True, ignore_status=True)
+            if disk.is_dir_mounted(self.dir):
+                return False
+            return True
+
+        self.log.info("checking if disk is mounted.")
+        if disk.is_disk_mounted(l_disk):
+            self.log.info("%s is mounted, unmounting it ....", l_disk)
+            if wait.wait_for(is_disk_unmounted, timeout=10):
+                self.log.info("%s unmounted successfully" % l_disk)
+            else:
+                self.err_mesg.append("%s unmount failed", l_disk)
+        else:
+            self.log.info("disk %s not mounted." % l_disk)
+        self.log.info("checking if dir %s is mounted." % self.dir)
+        if disk.is_dir_mounted(self.dir):
```

Comment:
How is this check different from previous one is_disk_mounted()?

Reply:
> How is this check different from previous one is_disk_mounted()?

we may not use the same disk all the time in our testing. But we use the same sw_raid_name, LV, VG names.
So if user passes different disk name then previously used disk which is mounted on same dir will be still there, and it will be same case again, like disk/dir in use and doesn't proceed further. Hence if we use the dir , this situation can be taken care.